src/ChannelServer/Network/Sending/Send.Parties.cs

```diff
@@ -0,0 +1,447 @@
+        /// </summary>
+        /// <param name="creature"></param>
+        /// <param name="party"></param>
+        public static void PartyLeaveUpdate(Creature creature, MabiParty party)
+        {
+            var packet = new Packet(Op.PartyLeaveUpdate, 0);
+            packet.PutLong(creature.EntityId);
+            
+            party.Broadcast(packet, true);
+        }
+
+        /// <summary>
+        /// Informs party members of a party type change (Dungeon, Normal, Jam, etc)
+        /// </summary>
+        /// <param name="party"></param>
+        public static void PartyTypeUpdate(MabiParty party)
+        {
+            var packet = new Packet(Op.PartyTypeUpdate, 0);
+
+            packet.PutInt((int)party.Type);
+
+            party.Broadcast(packet, true);
+        }
+
+        /// <summary>
+        /// Updates members on changes to the party settings
+        /// </summary>
+        /// <remarks>(apparently they only get to know about name changes?)</remarks>
+        /// <param name="party"></param>
+        public static void PartySettingUpdate(MabiParty party)
+        {
+            var packet = new Packet(Op.PartySettingUpdate, 0);
+            packet.PutString(party.Name);
+
+            party.Broadcast(packet, true);
+        }
+
+        /// <summary>
+        /// Response to the leader changing party settings.
+        /// </summary>
+        /// <param name="creature"></param>
+        public static void PartyChangeSettingR(Creature creature)
+        {
+            var packet = new Packet(Op.PartyChangeSettingR, creature.EntityId);
+
+            BuildPartyInfo(creature.Party, ref packet);
+            creature.Client.Send(packet);
+        }
+
+        /// <summary>
+        /// Responding whether the leader can (did) or can't (didn't) remove the requested party member.
+        /// </summary>
+        /// <param name="creature"></param>
+        /// <param name="canRemove"></param>
+        public static void PartyRemoveR(Creature creature, bool canRemove)
+        {
+            var packet = new Packet(Op.PartyRemoveR, creature.EntityId);
+            packet.PutByte(canRemove);
+
+            creature.Client.Send(packet);
+        }
+
+        /// <summary>
+        /// Informing the leader on the status of their password request
+        /// </summary>
+        /// <param name="creature"></param>
+        public static void PartyChangePasswordR(Creature creature)
+        {
+            var packet = new Packet(Op.PartyChangePasswordR, creature.EntityId);
+
+            // Changing password success, assume always successful?
+            packet.PutByte(true);
+
+            creature.Client.Send(packet);
+        }
+
+        /// <summary>
+        /// Informs all members of a change in leadership.
+        /// </summary>
+        /// <param name="party"></param>
+        public static void PartyChangeLeader(MabiParty party)
+        {
+            var packet = new Packet(Op.PartyChangeLeaderUpdate, 0);
```

Comment:
I don't have any logs at hand right now, but do all those packets really use id 0? I would've expected it to be the leader, or the creature doing the action, or the channel id.